src/data-types-lib/fieldTypes/string/index.js

```diff
@@ -48,6 +56,30 @@ export default {
       code: ERROR_CODE_VALIDATION,
       id: ERROR_MAX_EXCEEDED,
       message: param
+    }),
+
+    [CONSTRAINT_EMAIL]: param => param && (
+      // eslint-disable-next-line max-len
+      /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.
+        test(value.toLowerCase()
+        ) || throwError({
+        code: ERROR_CODE_VALIDATION,
+        id: ERROR_INVALID_EMAIL
+      })),
+
+    [CONSTRAINT_URL]: param => param && (
+      // eslint-disable-next-line max-len
+      /^(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i.
+        test(value.toLowerCase()
```

Comment:
At least in theory, [URLs are case-sensitive](https://stackoverflow.com/a/7996997)